components/availability/src/Availability.svelte

```diff
@@ -139,33 +142,99 @@
     });
   //#endregion layout
 
-  function handleTimeSlotClick(selectedSlot: any): string {
-    if (selectedSlot.selectionStatus === "unselected") {
-      if (click_action === "choose") {
-        sendTimeSlot(selectedSlot);
-      }
-      slotSelection = [...slotSelection, selectedSlot];
-      return "selected";
-    } else {
-      slotSelection = slotSelection.filter(
-        (chosenSlot) => chosenSlot != selectedSlot,
-      );
+  let eventOrganizer: string = "";
+  let eventParticipant: array = []; // email, name, status
+  let consecutiveSlotsEndTime: Date;
+  $: consecutiveSlotsEndTime;
 
-      return "unselected";
-    }
+  $: if (sortedSlots.length > 1) {
+    consecutiveSlotsEndTime = qetConsecutiveSlotsEndTime(sortedSlots);
+  }
+
+  //#region event query
+  let query: Availability.EventQuery;
+  $: query = {
+    component_id: id,
+    access_token: access_token,
+    calendarIDs: [],
+  };
+  //#region event query
+
+  //#region book consecutive slots as one event
+  function sendTimeSlot(timeSlot: Availability.TimeSlot) {
+    dispatchEvent("timeSlotChosen", { timeSlot });
   }
 
-  function sendTimeSlot(selectedSlot: Availability.TimeSlot) {
-    let start_time = new Date(selectedSlot.start_time);
-    let end_time = new Date(selectedSlot.end_time);
-    const timeslot: Availability.TimeSlot = {
-      start_time,
-      end_time,
+  function setTimeSlot(slots: obj[]): Availability.TimeSlot {
+    const consecutiveSlots = hasConsecutiveSlots(slots);
+    return {
+      start_time: slots[0].start_time,
+      end_time: consecutiveSlots ? consecutiveSlotsEndTime : slots[0].end_time,
     };
-    dispatchEvent("timeSlotChosen", {
-      timeslot,
-    });
   }
+
+  function getSlotsDuration(slots: obj[]): number {
+    return slot_size * slots.length;
+  }
+
+  function qetConsecutiveSlotsEndTime(slots: obj[]) {
+    const startTime = new Date(slots[0].start_time);
+    const slotsDuration = getSlotsDuration(slots);
+    return new Date(
+      startTime.setMinutes(startTime.getMinutes() + slotsDuration),
+    );
+  }
+
+  function hasConsecutiveSlots(slots: obj[]): boolean {
+    if (slots.length <= 1) {
+      return false;
+    } else {
+      const lastSlotEndTime = slots[slots.length - 1].end_time.valueOf();
+      const lastConsecutiveSlotEndTime = consecutiveSlotsEndTime.valueOf();
+      return lastSlotEndTime == lastConsecutiveSlotEndTime ? true : false;
+    }
+  }
+  //#endregion book consecutive slots as one event
+
+  //#region booking event logic for single time slot or consecutive time slots
+  function setSlotsSelectedStatus(slots: obj[]) {
+    slots.forEach((slot) => (slot.selectionStatus = "selected"));
+  }
+
+  function resetSlotSelection(slots: obj[]): [] {
+    return (slots = []);
+  }
+
+  function createEventFromTimeSlot(
+    event: Availability.TimeSlot,
+    query: Availability.eventQuery,
+  ) {
+    EventStore.createEvent(event, query);
```

Comment:
@heathergh I'm getting a `error: "Invalid 'calendar_id': "` response when attempting to book a time slot. Is this expected to work?

Reply:
No, the story for passing calendar IDs is not finished yet: https://app.clubhouse.io/nylas/story/63692/availability-viewer-accept-calendar-ids-to-automatically-read-available-timeslots